docs/content/direct/observability.md

```diff
@@ -0,0 +1,63 @@
+
+## Metrics Endpoints
+
+KubeStellar controllers expose Prometheus-compatible metrics endpoints. These endpoints respond to HTTP requests for metrics and can be queried by any monitoring system; KubeStellar does not mandate how metrics are collected or scraped.
+
+### Enabling Metrics
+
+KubeStellar controllers expose metrics at the `/metrics` endpoint by default. You may use any monitoring stack you prefer to collect these metrics.
+
+For an example of collecting both metrics and debug endpoint data using Prometheus, see the [monitoring/README.md](https://github.com/kubestellar/kubestellar/tree/main/monitoring/README.md). This is just one possible approach; KubeStellar does not require or mandate any specific monitoring tool or method.
+
+### Metrics Endpoint Table
+
+| Controller                      | Protocol | Port  | Path      | AuthN/AuthZ | Notes |
+|----------------------------------|----------|-------|-----------|-------------|-------|
+| kubestellar-controller-manager   | HTTPS    | 8443  | /metrics  | Kubernetes client authentication required when using the Service | Service: `kubestellar-controller-manager-metrics-service` |
+| ks-transport-controller         | HTTP     | 8090  | /metrics  | None (in-cluster) |  |
+| status-addon-controller         | HTTP     | 9280  | /metrics  | None (in-cluster) |  |
+| status-agent-controller         | HTTP     | 8080  | /metrics  | None (in-cluster) |  |
+
+> **Note:** When using the Service for `kubestellar-controller-manager`, protocol is HTTPS and Kubernetes client authentication is required. Direct pod access may use HTTP.
+
+## Debug/Profiling Endpoints
+
+Some KubeStellar components expose Go's built-in pprof debug endpoints for profiling and troubleshooting.
+
+### pprof Endpoint Table
+
+| Controller                      | Protocol | Port  | Path            | AuthN/AuthZ | Notes |
+|----------------------------------|----------|-------|-----------------|-------------|-------|
+| kubestellar-controller-manager   | HTTP     | 8082  | /debug/pprof/   | None        |  |
+| ks-transport-controller         | HTTP     | 8092  | /debug/pprof/   | None        |  |
+| status-addon-controller         | HTTP     | 9282  | /debug/pprof/   | None        |  |
+| status-agent-controller         | HTTP     | 8082  | /debug/pprof/   | None        |  |
+
+## Example: Accessing Metrics and Debug Endpoints
+
+> **Note:** The following example assumes you have a running KubeStellar controller pod and access to the appropriate Kubernetes context and namespace. Adjust the context, namespace, and deployment name as needed for your environment.
+
+```sh
+kubectl --context kind-kubeflex port-forward -n wds1-system deployment/kubestellar-controller-manager 8443:8443 8082:8082
+```
+- Access metrics: [https://localhost:8443/metrics](https://localhost:8443/metrics) (Kubernetes client authentication required)
```

Comment:
The rendering here is not what you intend, I think. You need a clearer paragraph break here, such as an empty line.

<img width="787" alt="image" src="https://github.com/user-attachments/assets/c15cd2a0-2514-4d1d-aea3-8e4edd3504b9" />